src/main/java/com/kodilla/ecommercee/service/ProductService.java

```diff
@@ -0,0 +1,36 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.Product;
+import com.kodilla.ecommercee.repository.ProductRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ProductService {
+
+    @Autowired
+    private ProductRepository productRepository;
+
+    public List<Product> getAllProducts() {
+        return productRepository.findAll();
+    }
+
+    public Product getProductById (final long id) {
+        return productRepository.findById(id).orElse(null);
+    }
+
+    public Product saveProduct(final Product product) {
+        return productRepository.save(product);
+    }
+
+    public Optional<Product> getProduct(final Long id) {
+        return productRepository.findById(id);
+    }
+
+    public void deleteProduct(final Long id) {
```

Comment:
można by było tutaj sprawdzić czy to co usuwamy jeszcze istnieje w bazie, a jeżeli tego nie ma to wyrzucić wyjątek

Reply:
Zmieniłem. Zrobiłem tak, jak wydawało mi się, że będzie ładnie, choć nadal nie do końca mi się to podoba... Albo jest nie do końca ładnie, albo w ogóle bez sensu :)